### DISCUSSION TITLE
네이밍 컨벤션에 대한 논의

### DISCUSSION THREAD

**Discussion by kyungjunleeme** [8percent/styleguide#23] [General]
- 플랫폼 이용료 일정산 작업 과정에서 일정산을 하기위한 플랫폼 이용료를 `UserPaid` 로부터 추출해서, 별도의`DailyCommission`, `DailyCommissionDetail` 에 일단위로 데이터를 기록하고 있습니다. 이 과정을 진행하던 중 함수명 네이밍에 어려움을 겪었습니다. 그래서 이번 계기를 기회로 발판삼아, 네이밍 컨벤션에 대해서 논의해보고 싶습니다.

- action 과 entity를 활용해서 함수명을 지을 때 , 순서를 어떻게 하면 좋을까요? 

# option 1  - user_create

# option 2 - create_user

참고 자료 및 관련 PR
- https://github.com/8percent/eight/pull/12867
- https://github.com/HackSoftware/Django-Styleguide#naming-convention

**Comment by Kirade** [comment]
저는 Option 2 가 적절하다고 생각했습니다.

Option 1 의 장점을 살펴보면 장점으로 Namespacing 과 Greppability 를 소개하고 있습니다.
만약 저희가 엔티티 개념을 차용해서 엔티티를 관리하고 있다면 고민해 볼만한 옵션이 될 수 있을것 같습니다.

하지만 현재 저희들은 엔티티로 도메인을 관리하지 않아서 위의 장점들이 생각보다 유용하지 않을것 같고, 코드 읽기가 직관적이지 않다는 점에서 별로인것 같다 생각했습니다.

**Comment by leehosung** [comment]
저도 option2 에 한표를 던집니다. 

특히 신규 프로젝트가 아닌 현재의 프로젝트의 경우에는 모든 함수의 명을 위 규칙에 따라서 변경하거나, 상당기간동안 양쪽을 유지하는 것에 드는 비용에 비해 
새롭게 얻어지는 가치가 크지 않다고 생각 합니다. 

대체로 저희가 접근할 수 있는 대부분의 프로젝트들이 action 을 먼저 두는 방식을 쓰고 있기 때문에, 익숙하다는것도 큰 이유가 되겠습니다.

**Comment by kyungjunleeme** [comment]
저는 개인적으로, 입사 전에는 네이밍 컨벤션에 큰 의미를 부여하지 않았습니다. 아마 혼자 개발을 주로 했기 때문이어서, 필요성과 중요성을 많이 느끼지 못했던것 같습니다. 하지만, 입사 후 코드리뷰 및 향후 유지보수를 위해서 네이밍 컨벤션이 중요하다는 것을 알게되었습니다. 저는 option 1, option2 둘 중에 뭐가 좋은지 판단하기가 힘든 것 같습니다. 다수결에 따르도록 하겠습니다. 
>  https://github.com/StyleShare/swift-style-guide   이 경우는 option1이 사용된 것 같네요.

**Reply by cbscsm** [reply]
해당 Repo가 많이 쓰이는 컨벤션인지 잘 모르겠습니다. 다소 독특한 컨벤션이네요. 특히 액션 함수에 대한 컨벤션은 문법적으로도 이상해서 과연 글로벌하게 통용되는 방식일지 의문입니다. 스위프트 언어의 특수성 때문인지는 모르겠네요.

**Comment by cbscsm** [comment]
저도 option2에 한표입니다. 일반적인 함수 네이밍 컨벤션이 그러하고, namespacing이 필요한 경우라면 클래스나 모듈 등 다른 방법도 있기 때문에 함수 이름의 일관성을 깨면서 사용할 필요는 없다고 생각됩니다.
